### PR TITLE
Feature refresh control

### PR DESCRIPTION
**Pull request by netbe**
Custom Refresh Control that can be added on a UIScrollview based class. i.e. UICollectionView

tasks:

* [x] Example
* [x] replace case when no custom image is given. Custom images to be bundled or custom View that can be rotated
* [x] Code review

### CODE REVIEW

File: Example/S2MToolbox/S2MTextLoadingView.m [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+#pragma mark - <S2MControlLoadingView>
+
+- (void)startAnimating
+{
+    [UIView animateWithDuration:0.3 animations:^{
+        self.label.alpha = 1;
+    }];
+
+    self.timer = [NSTimer scheduledTimerWithTimeInterval:1 target:self selector:@selector(timerFires:) userInfo:nil repeats:YES];
```
Comment (netbe) [inline Example/S2MToolbox/S2MTextLoadingView.m]: @joernito this class is not part of the toolbox just an example and has no intention to be reused. so I would leave it as it is

File: Example/S2MToolbox/S2MTextLoadingView.m [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+- (void)stopAnimating
+{
+    [self.timer invalidate];
+    self.label.textColor = [UIColor whiteColor];
```
Comment (netbe) [inline Example/S2MToolbox/S2MTextLoadingView.m]: same remark 
> this class is not part of the toolbox just an example and has no intention to be reused. so I would leave it as it is

File: Example/S2MToolbox/S2MTextLoadingView.m (hunk 1 above)
Comment (joernito) [inline Example/S2MToolbox/S2MTextLoadingView.m]: ok.. fine with that

File: UIKit/S2MRefreshControl.m [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        id newDelegate = change[NSKeyValueChangeNewKey];
+        self.originalScrollViewDelegate = newDelegate;
+    }
+}
+
+
+#pragma mark - UIScrollViewDelegate
+//// We forward delegate calls
+
+- (BOOL)respondsToSelector:(SEL)aSelector
```
Comment (netbe) [inline UIKit/S2MRefreshControl.m]: yes it is. see https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtForwarding.html

**Comment by netbe** [comment]
@joernito I adjusted code following the comments...

**Comment by joernito** [comment]
looks great!

**Pull request merged**